include/alpaka/pltf/PltfGenericSycl.hpp

```diff
@@ -28,150 +28,95 @@ namespace alpaka
+                      }
+                      throw std::runtime_error(ss_err.str());
+                  }}}
         {
-            auto lock = std::scoped_lock<std::mutex>{mutex};
-
-            if(!initialized)
-                do_initialization();
-
-            return *platform_opt;
         }
 
-        [[nodiscard]] static auto syclDevices() -> std::vector<sycl::device>&
+        [[nodiscard]] auto syclPlatform() -> sycl::platform&
         {
-            auto lock = std::scoped_lock<std::mutex>{mutex};
-
-            if(!initialized)
-                do_initialization();
-
-            return devices;
+            return platform;
         }
 
-        [[nodiscard]] static auto syclContext() -> sycl::context&
+        [[nodiscard]] auto syclDevices() -> std::vector<sycl::device>&
         {
-            auto lock = std::scoped_lock<std::mutex>{mutex};
-
-            if(!initialized)
-                do_initialization();
-
-            return *context_opt;
+            return devices;
         }
 
-        static auto initialize() -> void
+        [[nodiscard]] auto syclContext() -> sycl::context&
         {
-            auto lock = std::scoped_lock<std::mutex>{mutex};
-            do_initialization();
-        }
-
-        static auto reset() -> void
-        {
-            auto lock = std::scoped_lock<std::mutex>{mutex};
-
-            if(!initialized)
-                return;
-
-            context_opt.reset();
-            devices.clear();
-            platform_opt.reset();
-            initialized = false;
+            return context;
         }
 
     private:
-        static auto do_initialization()
-        {
-            if(initialized)
-                return;
-
-            platform_opt = sycl::platform{TSelector{}};
-            devices = platform_opt->get_devices();
-            context_opt = sycl::context{
-                devices,
-                [](sycl::exception_list exceptions)
-                {
-                    auto ss_err = std::stringstream{};
-                    ss_err << "Caught asynchronous SYCL exception(s):\n";
-                    for(std::exception_ptr e : exceptions)
-                    {
-                        try
-                        {
-                            std::rethrow_exception(e);
-                        }
-                        catch(sycl::exception const& err)
-                        {
-                            ss_err << err.what() << " (" << err.code() << ")\n";
-                        }
-                    }
-                    throw std::runtime_error(ss_err.str());
-                }};
-
-            initialized = true;
-        }
-
-#    if BOOST_COMP_CLANG
-#        pragma clang diagnostic push
-#        pragma clang diagnostic ignored "-Wexit-time-destructors"
-#    endif
-        inline static std::mutex mutex;
-        inline static bool initialized{false};
-
-        inline static std::optional<sycl::platform> platform_opt{std::nullopt};
```

Comment:
Since we're removing `std::optional` we can also remove the corresponding header, no?

Reply:
Right.